### ISSUE TITLE
Support Integration Tests for training

### ISSUE THREAD

**Issue by YongseopKim** [Samsung/ONE#12481]
Let's support Integration Tests for training.
- circle model -> train -> compare with tf model

To get it, we have to clarify 
1. How to validate? How to realize the model is trained?
    - Fixed/Static random seed and golden data?
2. How many time to test? Training takes much more cost than inference. 

/cc @Samsung/one_onert

**Comment by jyoungyun** [comment]
### How to validate

1. Prepare datasets
2. Split the dataset into train, valid, and test datasets (valid dataset is optional)
3. Test original model using test datasets <- **before**
4. Train model using train dataset
5. (If there is valid datasets) While training, check validation loss and accuracy (I'm not sure how this data will affect us 👀)
6. Test updated model using test datasets <- **after**
7. Compare **before** and **after** results
   1. If the accuracy has decreased, it fails
   2. If the accuracy has increased, it succeeds

How about this scenario? :)

**Comment by ragmani** [comment]
> Fixed/Static random seed and golden data?

If there any way to do integration tests with random data to check training was successful, you can use random data for it. But I couldn't find any way for it.

> How many time to test? Training takes much more cost than inference.

I guess it depends a lot on what model it is. 
IMHO, for PR test, within 5-10 minutes. For daily test within 1-2 time.

**Comment by ragmani** [comment]
> 7. Compare **before** and **after** results
   1. If the accuracy has decreased, it **fail**s
   2. If the accuracy has increased, it **succeed**s

It doesn't make sense just to check whether accuracy increases. I think it may be too loose tests.
If we use fixed datasets, I think we can consider checking over specific accuracy or checking degree of increase in accuracy.

**Comment by YongseopKim** [comment]
> Compare before and after results
>  If the accuracy has decreased, it fails
>  If the accuracy has increased, it succeeds

IHMO, this could be naive way.

I'd like to focus our goal that is to check integration. (This means that the better/lower loss-value than tf's one is not high priority now.)

If we support such as [these initalizers in tf/keras](https://www.tensorflow.org/api_docs/python/tf/keras/initializers/RandomNormal), we could get fixed loss-values. Then, we can test the integration.

> IMHO, for PR test, within 5-10 minutes. For daily test, within 1-2 hours.

Agree.

**Comment by jyoungyun** [comment]
> I'd like to focus our goal that is to check integration. (This means that the better/lower loss-value than tf's one is not high priority now.)

I agree. Currently, there is no CI test for the actual operation according to the code change. Thus, this alone is worth it.

If we need to check the accuracy, we also can use TensorFlow results. I usually compared the TensorFlow results with ours to verify our training operation in my local environment.

Do we have to use random data? What is the advantage of using random data instead of using well-known datasets for integration test?

**Comment by YongseopKim** [comment]
> Just out of curiosity, do we have to use random data? What is the advantage of using random data instead of using well-known datasets for integration test?

Hmm... there are a bit of confusing. Random seed something like that which I talked is not about input.

And the `initalizer` that I mentioned above is for init variable(weight to be trained).

My focusing point is that. 😄

**Comment by YongseopKim** [comment]
As I and @jyoungyun discussed, we talked
- Tests focusing on a unit **model** not **op**
- (In addition) perf feature would be great if introduce
